server/src/main/java/com/linecorp/centraldogma/server/metadata/MetadataService.java

```diff
@@ -407,8 +407,11 @@ public CompletableFuture<Revision> updatePerRolePermissions(Author author,
         requireNonNull(perRolePermissions, "perRolePermissions");
 
         if (Project.isReservedRepoName(repoName)) {
-            throw new UnsupportedOperationException(
-                    "can't update the per role permission for internal repository: " + repoName);
+            final Set<Permission> guest = perRolePermissions.guest();
+            if (!guest.isEmpty()) {
+                throw new UnsupportedOperationException(
+                        "can't give a permission to guest for internal repository: " + repoName);
```

Comment:
Since I believe this exception will be directly exposed to the end user

```suggestion
                        "Can't give a permission to guest for internal repository: " + repoName);
```